### PR TITLE
Fixed sonar issues

### PR DESCRIPTION
**Pull request by nilasini**
### Proposed changes in this pull request

[List all changes you want to add here. If you fixed an issue, please
add a reference to that issue as well.]

-

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementService.java [hunk 1]
```diff
@@ -181,6 +181,7 @@ public abstract String getServiceProviderNameByClientId(String clientId, String
     public abstract ServiceProvider getServiceProvider(String serviceProviderName, String tenantDomain)
             throws IdentityApplicationManagementException;
 
+    @Deprecated
```
Comment (isharak) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementService.java]: Need to add a comment, If there is an alternative specify that

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java [hunk 2]
```diff
@@ -745,6 +745,7 @@ public ServiceProvider getServiceProvider(String serviceProviderName, String ten
      * @return
      * @throws IdentityApplicationManagementException
      */
+    @Deprecated
```
Comment (isharak) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java]: Need to add a comment, If there is an alternative specify that

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java [hunk 3]
```diff
@@ -116,23 +116,26 @@ private static Registry getRegistryForAnonymousSession(String domainName, String
...
+                domainName = MultitenantConstants.SUPER_TENANT_DOMAIN_NAME;
+            }
+
+            return AnonymousSessionUtil.getRealmByTenantDomain(registryService, realmService,
+                    domainName);
+
         } catch (CarbonException e) {
-            log.error("Error obtaining the realm", e);
-            throw IdentityException.error("Error Obtaining a realm", e);
+            log.error("Error Obtaining a realm for user name: " + username + " and " +
```
Comment (hasinthaindrajee) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java]: Do not log and throw..

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java [hunk 4]
```diff
@@ -755,21 +755,17 @@ public ServiceProvider getServiceProvider(int appId) throws IdentityApplicationM
         ApplicationDAO appDAO = ApplicationMgtSystemConfig.getInstance().getApplicationDAO();
         ServiceProvider serviceProvider = appDAO.getApplication(appId);
 
+        if (serviceProvider == null){
```
Comment (madurangasiriwardena) [inline components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java]: Formatting issue.

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/IdentityClaimManager.java [hunk 5]
```diff
@@ -91,35 +91,30 @@ public static IdentityClaimManager getInstance() throws IdentityException {
      */
     public Claim[] getAllSupportedClaims(String dialectUri, UserRealm realm)
             throws IdentityException {
-        Claim[] claims = new Claim[0];
+
         ClaimManager claimAdmin;
-        ArrayList<Claim> requiredClaims = null;
+        ArrayList<Claim> requiredClaims = new ArrayList<Claim>();
```
Comment (madurangasiriwardena) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/IdentityClaimManager.java]: Better to use List interface when creating variable and use diamond operator when initializing variable.

File: components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java (hunk 4 above)
Comment (nilasini) [reply components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationManagementServiceImpl.java]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/1303

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/IdentityClaimManager.java (hunk 5 above)
Comment (nilasini) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/IdentityClaimManager.java]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/1303

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/bean/context/MessageContext.java [hunk 6]
```diff
@@ -30,7 +30,7 @@
     protected Map<T1,T2> parameters = new HashMap<>();
 
     public MessageContext(Map<T1,T2> parameters){
-        if(parameters != null || MapUtils.isNotEmpty(parameters)) {
+        if(MapUtils.isNotEmpty(parameters)) {
```
Comment (pulasthi7) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/bean/context/MessageContext.java]: Fix formatting

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java [hunk 7]
```diff
@@ -260,7 +261,7 @@ public static String getTenantDomain(int tenantId) throws IdentityRuntimeExcepti
                     tenantId + e.getMessage(), e);
         }
         if(tenantDomain == null){
```
Comment (pulasthi7) [inline components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java]: Fix formatting

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/bean/context/MessageContext.java (hunk 6 above)
Comment (nilasini) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/bean/context/MessageContext.java]: fixed with https://github.com/wso2/carbon-identity-framework/pull/1319

File: components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java (hunk 7 above)
Comment (nilasini) [reply components/identity-core/org.wso2.carbon.identity.core/src/main/java/org/wso2/carbon/identity/core/util/IdentityTenantUtil.java]: Fixed with https://github.com/wso2/carbon-identity-framework/pull/1319

**Pull request merged**